packages/portal/src/components/embed/EmbedGateway.vue

```diff
@@ -253,11 +259,25 @@
 
       openModalOrSaveConsents() {
         if (this.cookieConsentRequired) {
-          this.$bvModal.show('cookie-modal');
+          this.openMainCookieModalAndScrollToThirdPartyContent();
         } else {
           this.klaroManager?.saveAndApplyConsents('save');
           this.checkConsentAndOpenEmbed();
         }
+      },
+
+      async openMainCookieModalAndScrollToThirdPartyContent() {
+        this.$bvModal.show('cookie-modal');
+        await this.$nextTick();
+
+        // Wait for b-modal to be ready
+        setTimeout(() => {
```

Comment:
Could/should we use the waitFor utility to ensure the availability of the element, and also not wait longer than necessary for it?